### ISSUE TITLE
runtest.sh failure on Mac OSX 10.12.4

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#10680]
When I run tests\runtest.sh, I see:
```
/Users/bruce/gh/coreclr/tests/runtest.sh: line 375: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/Users/bruce/gh/coreclr/tests/runtest.sh: line 379: mscorlib.dll: syntax error: invalid arithmetic operator (error token is ".dll")
```
This is in the following code:
```
function precompile_overlay_assemblies {
    declare -A skipCrossGenFiles

    while read line
    do
      skipCrossGenFiles[$line]=1
    done <  "$(dirname "$0")/skipCrossGenFiles.$ARCH.txt"
...
```
@sdmaclea You added this code in https://github.com/dotnet/coreclr/pull/9907. Any suggested fix? (I'm not a Unix/Mac shell script guru by any means...)

(Does the CI see this? If not, why not?)

cc @wtgodbe

**Comment by sdmaclea** [comment]
@BruceForstall "declare -A " creates a hash map in bash version 4.  I am not sure why this went through CI, but at one point I thought the mac osx 10.12 leg was disabled.  Bash version 3 did not support hash map.

I will try to prepare a patch to fix this, but unfortunately I do not have an older bash to test against.

**Comment by sdmaclea** [comment]
I think --crossgen option is rarely used.  Perhaps it is only used when running r2r legs.

**Comment by danmosemsft** [comment]
@sdmaclea is this still an issue? seems you fixed it? (i didn't try)

**Comment by sdmaclea** [comment]
@danmosemsft This can be closed it was fixed by #10697
